tensorboard/components/tensor_widget/selection.ts

```diff
@@ -225,26 +221,25 @@ export class TensorElementSelection {
    * Moving a multi-element selection always causes the selection to
    * collapse to a single element.
    *
+   * If the slicing spec has changed as a result of this move, and the
+   * `onSlicingSpecChange` callback is specified, the callback will be invoked.
+   *
    * @param direction Direction in which this movement is being made.
-   * @return If the selection movement doesn't necessitate an update to the
-   *   slicing spec (i.e., if the new single element selection is within
-   *   the current slicing spec), `null`. Else, a new `TensorViewSlicingSpec`
-   *   object that can accommodate the new single-element selection.
    */
-  public move(direction: SelectionMoveDirection): TensorViewSlicingSpec | null {
-    // let viewRangeChanged = false;
-    let newSlicingSpec: TensorViewSlicingSpec | null = null;
+  public move(direction: SelectionMoveDirection): void {
+    let viewRangeChanged = false;
+    // let newSlicingSpec: TensorViewSlicingSpec | null = null;
```

Comment:
Per offline discussion: reinstate this, as a copy of a slicingSpec passed to move(), and remove this.slicingSpec entirely.  Then returning newSlicingSpec is fine after all; no need for the callback either.  Thanks and sorry again for going in circles :)

Reply:
@davidsoergel I made a more thorough clean up: `move()` doesn't really return a slicingSpec anymore. It returns a MoveDirection enum to tell the caller which direction the slicingSpec should move. It's up to the caller to actually update the slicingSpec. The caller doesn't really have to update it.

There gets rid of the unclear ownership, removes the need to copy things and makes testing easier.

Reply:
Sweet